shell/components/nav/WindowManager/ContainerShell.vue

```diff
@@ -282,16 +278,44 @@ export default {
       this.socket.addEventListener(EVENT_DISCONNECTED, (e) => {
         this.isOpen = false;
         this.isOpening = false;
+
+        // If we had an error message, try connecting with the next command
+        if (this.errorMsg) {
+          this.terminal.write(this.errorMsg);
+          if (this.backupShells.length && this.retries < 2) {
+            this.retries++;
+            // we're not really counting on this being a reactive change so there's no need to fire the whole action
+            this.pod.os = undefined;
+            // the pod will still return an os if one's been defined in the node so we'll skip the backups if that's the case and rely on retry count to break the retry loop
+            if (!this.pod.os) {
+              this.os = this.backupShells.shift();
```

Comment:
Now I'm wondering if the design of using `shift` is to purposefully deplete all members of `this.backupShells`? It's not immediately clear to me.

Reply:
Yup! That's the design, we're effectively draining backupShells as we try them until we run out.